include/onnxruntime/interface/graph/graph.h

```diff
@@ -0,0 +1,190 @@
+  /// representing the dimensions of the value. Use -1 for unknown dimensions.
+  /// </returns>
+  virtual std::optional<std::vector<int64_t>> Shape() const = 0;
+
+  /// <returns>The inferred/declared dtype of the value. UNDEFINED (0) if dtype is unknown.</returns>
+  virtual DataType DType() const = 0;
+
+  virtual ~ValueInfoViewRef(){};
+};
+
+class NodeViewRef {
+ public:
+  virtual size_t Index() const = 0;
+
+  /// <returns>Op computed by the node</returns>
+  virtual std::string_view OpType() const = 0;
+
+  /// <returns>Domain containing the op. Empty string if node has no domain set.</returns>
+  virtual std::string_view Domain() const = 0;
+
+  /// <returns>Names of input values. Empty string may be included for optional inputs.</returns>
+  virtual std::vector<std::string_view> Inputs() const = 0;
+
+  /// <returns>Names of output values. Empty string may be included for optional outputs.</returns>
+  virtual std::vector<std::string_view> Outputs() const = 0;
+
+  /// <param name="name">Name of the attribute to return</param>
+  /// <returns>
+  /// The attribute value, or nullopt if the attribute is not present on the node, or is not of type int.
+  /// </returns>
+  virtual std::optional<int64_t> GetAttributeInt(std::string_view name) const = 0;
+
+  /// <param name="name">Name of the attribute to return</param>
+  /// <returns>
+  /// The attribute value, or nullopt if the attribute is not present on the node, or is not of type string.
+  /// </returns>
+  virtual std::optional<std::string> GetAttributeString(std::string_view name) const = 0;
+
+  /// <param name="name">Name of the attribute to return</param>
+  /// <returns>
+  /// The attribute value, or nullopt if the attribute is not present on the node, or is not of type int[].
+  /// </returns>
+  virtual std::optional<std::vector<int64_t>> GetAttributeInts(std::string_view name) const = 0;
+
+  /// <summary>
+  /// Convenience method. Returns whether node is of the specified op type and domain
+  /// </summary>
+  /// <param name="op_type">Op type</param>
+  /// <param name="domain">Domain. Empty string and "onnx.ai" are treated as equal.</param>
+  /// <returns></returns>
+  virtual bool IsOp(std::string_view op_type, std::string_view domain = "") const {
+    if (OpType() != op_type) {
+      return false;
+    }
+    std::string_view node_domain = Domain();
+    return node_domain == domain ||
+           ((domain == "" || domain == "ai.onnx") && (node_domain == "" || node_domain == "ai.onnx"));
+  }
+
+  /// <summary>
+  /// Convenience method. Returns value of int attribute with name, or given default if unset.
+  /// </summary>
+  /// <param name="name">Attribute name</param>
+  /// <param name="default_value">Default value</param>
+  /// <returns>Attribute value or default value</returns>
+  virtual int64_t GetAttributeIntDefault(std::string_view name, int64_t default_value) const {
```

Comment:
>int64_t default_value [](http://example.com/codeflow?start=64&length=21)

I don't think we should add default attribute value support on Node. the "default attribute value" is a concept for onnx op, not on node.